git-migration/ssd_svn2git.sh

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env bash
+set -euo pipefail
+# This script
+# - Reads config file
+# - For each repository
+#   - Convert svn trunk to git repository in WORKDIR
+#   - Attach trunk release tags to git repository
+#   - Create remote repository (github.com/MetOffice/REPO_NAME), if necessary
+#   - Push local branch to remote
+#   - Push tags to remote
+
+usage(){
+  printf "%-s\n" "Usage: ${0##*/} [OPTION]
+
+  ${0##*/} Migrates the Met Office Science Repositories to GitHub.
+
+  Options:
+    -h, --help              Display this help and exit
+    -d, --debug             Enable command trace (debug mode)
+    -u, --update            Update GitHub repository.
+                            Please let the admin deal with this option.
+    -c, --config[=VALUE]    Path to config.json file (default: config.json)
+    -w, --workdir[=VALUE]   Working directory (default: ${DATADIR}/git-migration)
+
+  Examples:
+    ${0##*/} -c config.json -w ${DATADIR}/git-migration
+    ${0##*/} --config=config.json --workdir=${DATADIR}/git-migration
+
+  Report bugs to <yaswant.pradhan@metoffice.gov.uk>" | sed 's/^  //g'
+}
+
+OPTS=$(getopt \
+    -o hduc:w: \
+    -l help,debug,update,config:,workdir: \
+    -n "${0##*/}" -- "$@") || { usage; exit 1; }
+eval set -- "$OPTS"
+# ----------------------------------------------------------------------------
+SCRIPT_DIR=$(dirname -- "$(readlink -f -- "$0")")
+CONFIG_FILE="${SCRIPT_DIR}/config.json"
+WORKDIR="${DATADIR}/git-migration"
+UPDATE_GITHUB=0
+# ----------------------------------------------------------------------------
+while true; do
+  case "$1" in
+    -h |--help) usage; exit 0 ;;
+    -d |--debug) PS4='L${LINENO}: '; set -x; shift 1 ;;
+    -u |--update) UPDATE_GITHUB=1; shift 1 ;;
+    -c |--config) export CONFIG_FILE="$2"; shift 2 ;;
```

Comment:
The optional `CONFIG_FILE` should store the full path to the configuration file - something that is already done for the default - otherwise later portions of the script will fail.  It also means that the examples in the help will work.

```suggestion
    -c |--config) export CONFIG_FILE=$(realpath $2); shift 2 ;;
```